### ISSUE TITLE
Support throttle and debounce for selections

### ISSUE THREAD

**Issue by domoritz** [vega/vega-lite#5288]
It would be great to support throttle and debounce without having to modify the `on` selector.

cc @saulshanabrook

**Comment by arvind** [comment]
It'd be helpful for me to understand the motivation behind this issue -- why is specifying debouncing/throttling as part of the `on` property is insufficient? One thing I can think of is that the on property doesn't allow you to _augment_ the default selector (i.e., apply debouncing/throttling without changing the actual triggering events).

**Comment by saulshanabrook** [comment]
> One thing I can think of is that the on property doesn't allow you to augment the default selector (i.e., apply debouncing/throttling without changing the actual triggering events).

Yeah that is my use case. For example, to add a debounce to the default `on` and `translate` for the interval selection, I [have to do](https://nbviewer.jupyter.org/github/Quansight/jupyterlab-omnisci/blob/time/notebooks/Charting%20Example.ipynb) (in Altair):

```python
DEBOUNCE_MS = 400

dates = alt.selection_interval(
    fields=['dep_timestamp'],
    encodings=['x'],
    on=f'[mousedown, window:mouseup] > window:mousemove!{{0, {DEBOUNCE_MS}}}',
    translate=f'[mousedown, window:mouseup] > window:mousemove!{{0, {DEBOUNCE_MS}}}',
    zoom=False
)
```

**Comment by domoritz** [comment]
Besides being a bit cumbersome to use, having to use `on` property is also not very discoverable and hard to reason about programmatically.

**Comment by arvind** [comment]
> hard to reason about programmatically

Hmm, not sure I understand this. Besides the string selectors, we should also be supporting the [object definitions](https://vega.github.io/vega/docs/event-streams/#object). But, perhaps there's a bug that prevents objects from being specified?

**Comment by domoritz** [comment]
The object definition works but I doubt that anyone has ever done it because it's pretty tedious (I copied this from the generated Vega). You also have to know about the magic filter (although it doesn't seem to be necessary here).

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v3.json",
  "data": {"url": "data/cars.json"},
  "selection": {
    "brush": {
      "type": "interval",
      "init": {"x": [55, 160], "y": [13, 37]},
      "on": [{
        "source": "window",
        "type": "mousemove",
        "consume": true,
        "between": [
          {
            "source": "scope",
            "type": "mousedown",
            "filter": [
              "!event.item || event.item.mark.name !== \"brush_brush\""
            ]
          },
          {"source": "window", "type": "mouseup"}
        ]
      }]
    }
  },
  "mark": "point",
  "encoding": {
    "x": {"field": "Horsepower", "type": "quantitative"},
    "y": {"field": "Miles_per_Gallon", "type": "quantitative"},
    "color": {
      "condition": {"selection": "brush", "field": "Cylinders", "type": "ordinal"},
      "value": "grey"
    }
  }
}
```

I think having a simple `"debounce": 12` is much easier to read and reason about.

**Comment by arvind** [comment]
I meant that my preference would be to keep `debounce` within `on` (or, more generally, event stream definitions) since it's already there (rather than introduce it as part of a selection definition). Instead, we should be able to partially specify an `on` such that it gets merged together with existing/default definitions. This keeps things clean since event stream definitions can occur across multiple properties.

For example, replicating @saulshanabrook's example:

```json
"selection": {
  "dates": {"type": "interval", "on": {"debounce": 400}, "translate": {"debounce": 400}}
}
```

**Comment by domoritz** [comment]
Ahh, I see. That makes sense. I wonder, however, how one would distinguish overriding from merging. We could say that we always merge and you can set e.g. `"on": {"filter": null}` if you want to remove a default property. 

To go this route, we probably want to merge https://github.com/vega/vega-lite/pull/5290 first. I'll put the generator updates on my queue then.
